java/adapter/jdbc/src/main/java/org/apache/arrow/adapter/jdbc/JdbcToArrowUtils.java

```diff
@@ -528,4 +680,39 @@ private static void updateVector(VarBinaryVector varBinaryVector, Blob blob, boo
     updateVector(varBinaryVector, blob != null ? blob.getBinaryStream() : null, isNonNull, rowCount);
   }
 
+  private static void updateVector(
+      ListVector listVector,
+      ResultSet resultSet,
+      int arrayIndex,
+      int rowCount,
+      JdbcToArrowConfig config)
+      throws SQLException, IOException {
+
+    final JdbcFieldInfo fieldInfo = getJdbcFieldInfoForArraySubType(resultSet.getMetaData(), arrayIndex, config);
+    if (fieldInfo == null) {
+      throw new IllegalStateException("Column " + arrayIndex + " is an array of unknown type.");
+    }
+
+    final int valueCount = listVector.getValueCount();
+    final Array array = resultSet.getArray(arrayIndex);
+
+    FieldVector fieldVector = listVector.getDataVector();
+    int arrayRowCount = 0;
+
+    listVector.startNewValue(rowCount);
+
+    if (!resultSet.wasNull()) {
```

Comment:
If I had to guess at the var char problem, this doesn't look correct for handling nulls on arrays.   It seems that you want to explicitly set null instead of just skipping over the loop.  Looking at ListVector (quickly) I didn't see a method for doing this.  Looking at the java code-base it might be that the writer does have a way to do this (and might be better to use instead).  I.e. Somthing like listVector.getWriter() up top.  And then writeNull().  But I don't have a java dev environment setup so I can't give you specifics.  It might be worth asking about setting nulls on list vectors in general if pursuing this approach doesn't help.  (It might also pay to create a failing unit test if this is really causing issues.

Reply:
@mikepigott Could you please help me understand the issue that you are facing..

From my understanding there should not be a need to set nulls explicitly (since the validity buffer should be off to be begin with and is set to on as we set new values).

You can use the setPosition() method in the Writer to move the position to forward to ignore nulls from what i understand.

Reply:
Thanks @praveenbingo good to know.  We should probably try to document this someplace (a helper method that advances the position still seems potentially useful but maybe there are performance concerns)